app/routines/update_user_salesforce_info.rb

```diff
@@ -0,0 +1,148 @@
+class UpdateUserSalesforceInfo
+  USER_BLOCK_SIZE = 1000
+
+  def initialize
+    @errors = []
+  end
+
+  def self.call
+    new.call
+  end
+
+  def call
+    contacts_by_email = {}
+    contacts_by_id = {}
+
+    # Store a map of contacts by primary email
+
+    contacts.each do |contact|
+      next if contact.email.nil?
+      contacts_by_email[contact.email] = contact
+      contacts_by_id[contact.id] = contact
+    end
+
+    # Add contacts by alt email if they don't already exist in the map; error if they do.
+
+    contacts.each do |contact|
+      next if contact.email_alt.nil?
+
+      if contacts_by_email[contact.email_alt].present?
+        error!(message: "#{contact.email_alt} is an alt email on contact #{contact.id} but a " \
+                        "primary email on contact #{contacts_by_email[contact.email_alt].id}")
+      else
+        contacts_by_email[contact.email_alt] = contact
+        contacts_by_id[contact.id] = contact
```

Comment:
Good point.  I checked with Denver and he said primary email is required, but that there are some old Contacts that don't have it, so there is a (very low) chance that a Contact could have an alt email but not a primary.  So I might leave it (but with a comment)